src/nplinker/pairedomics/downloader.py

```diff
@@ -315,34 +318,33 @@ def _load_gnps_data(self, gnps_task_id) -> zipfile.ZipFile:
 
         # Try read from cache
         if os.path.exists(self.metabolomics_zip):
-            logger.info('Found existing metabolomics_zip at {}'.format(
-                self.metabolomics_zip))
+            logger.info('Found existing metabolomics_zip at %s',
+                        self.metabolomics_zip)
             try:
-                mbzip = zipfile.ZipFile(self.metabolomics_zip)
-                return mbzip
-            except zipfile.BadZipFile as bzf:
+                with zipfile.ZipFile(self.metabolomics_zip) as mbzip:
+                    return mbzip
+            except zipfile.BadZipFile:
                 logger.info(
                     'Invalid metabolomics zipfile found, will download again!')
                 os.unlink(self.metabolomics_zip)
         url = _generate_gnps_download_url(gnps_task_id)
         _execute_download(url, self.metabolomics_zip)
 
         # this should throw an exception if zip is malformed etc
-        mbzip = zipfile.ZipFile(self.metabolomics_zip)
-        return mbzip
-
+        with zipfile.ZipFile(self.metabolomics_zip) as mbzip:
```

Comment:
Same here, the original code is right, no need to change.

Reply:
Same as above, the original code is there now, but I disabled the pylint warning about consider-using-with